.devops/deploy-pipelines.yml

```diff
@@ -65,106 +72,102 @@ variables:
+                  displayName: 'Print variables'
                   inputs:
-                    workingDirectory: '$(Pipeline.Workspace)/helm'
                     targetType: 'inline'
                     script: |
-                      helm repo add pagopa-microservice https://pagopa.github.io/aks-microservice-chart-blueprint
-                      helm dep build
-                    failOnStderr: true
+                      echo "appVersion: $(appVersion)"
                 - bash: |
                     echo 'microservice-chart:
                        podAnnotations:
                           "build/buildNumber": "$(Build.BuildNumber)"
                           "build/appVersion": "$(appVersion)"
                           "build/sourceVersion": "$(Build.SourceVersion)"' > buildMetadata.yaml
+                    cat buildMetadata.yaml
+                    cat helm/values-${{ lower(variables.environment) }}.yaml
                   displayName: Writing build metadata
-
+                - task: Bash@3
+                  name: helm_dependency_build
+                  displayName: Helm dependency build
+                  inputs:
+                    workingDirectory: 'helm'
+                    targetType: 'inline'
+                    script: |
+                      helm repo add microservice-chart https://pagopa.github.io/aks-microservice-chart-blueprint
+                      helm dep build
+                    failOnStderr: true
+                - task: Bash@3
+                  name: helm_template
+                  displayName: Helm template
+                  inputs:
+                    workingDirectory: 'helm'
+                    targetType: 'inline'
+                    script: |
+                      helm template -f values-${{ lower(variables.environment) }}.yaml .
+                    failOnStderr: true
                 - task: HelmDeploy@0
-                  displayName: Helm upgrade
+                  displayName: Deploy on $(environment)
                   inputs:
                     kubernetesServiceEndpoint: ${{ variables.kubernetesServiceConnection }}
                     namespace: '$(deployNamespace)'
-                    command: upgrade
-                    chartType: filepath
-                    chartPath: $(Pipeline.Workspace)/helm
+                    command: 'upgrade'
+                    chartType: "filepath"
+                    chartPath: "helm"
                     chartName: ${{ variables.helmReleaseName }}
                     releaseName: ${{ variables.helmReleaseName }}
-                    valueFile: "$(Pipeline.Workspace)/helm/values-${{ lower(variables.environment) }}.yaml"
+                    valueFile: "helm/values-${{ lower(variables.environment) }}.yaml"
                     install: true
                     waitForExecution: true
-                    arguments: --timeout 5m0s
-                      --values buildMetadata.yaml
+                    arguments: --timeout 5m0s --debug
```

Comment:
`--debug` can be removed?

Reply:
Is better to live it, to allow to see problems when it arise